equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java

```diff
@@ -286,13 +286,15 @@ public Equipment update(@RequestBody Equipment equipment){
         } else if (existingApElementConfig != null && existingApElementConfig.isBlinkAllLEDs() != updatedApElementConfig.isBlinkAllLEDs()) {
             LOG.debug("Updated BlinkingLEDs {}", ret);
             event = new EquipmentBlinkLEDsEvent(ret);
-        } else if (equipment.getCustomerId() != existingEquipment.getCustomerId()) {
-            event = new EquipmentCustomerChangedEvent(existingEquipment, ret);
         } else {
             event = new EquipmentChangedEvent(ret);
         }
         publishEvent(event);
         
+        if (equipment.getCustomerId() != existingEquipment.getCustomerId()) {
```

Comment:
So if I understand the logic here, there could be 2 published events, one for whatever changed on the equipment, including customer, and another, for the customer itself changing. If that is the case, perhaps the CustomerChangedEvent should not be sent? The problem here is that the change may have been more than just customer, and it is being sent as a customerOnly change. 
IF it is only the customer that is being changed, make that your 1st conditional check, (i.e. replace 1st 'if' with that, and then make old 1st if an 'else if')